dask/array/tests/test_optimization.py

```diff
@@ -6,21 +6,95 @@
 
 import dask
 import dask.array as da
-from dask.array.core import getter, getter_nofancy
+from dask.array.chunk import getitem as da_getitem
+from dask.array.core import getter as da_getter
+from dask.array.core import getter_nofancy as da_getter_nofancy
 from dask.array.optimization import (
+    _is_getter_task,
     fuse_slice,
-    getitem,
     optimize,
     optimize_blockwise,
     optimize_slices,
 )
 from dask.array.utils import assert_eq
 from dask.highlevelgraph import HighLevelGraph
-from dask.optimization import fuse
+from dask.optimization import SubgraphCallable, fuse
 from dask.utils import SerializableLock
 
 
-def test_fuse_getitem():
+def _wrap_getter(func, wrap):
+    """
+    Getters generated from a Blockwise layer might be wrapped in a SubgraphCallable.
+    Make sure that the optimization functions can still work if that is the case.
+    """
+    if wrap:
+        return SubgraphCallable({"key": (func, "index")}, outkey="key", inkeys="index")
+    else:
+        return func
+
+
+@pytest.fixture(params=[True, False])
+def getter(request):
+    yield _wrap_getter(da_getter, request.param)
+
+
+@pytest.fixture(params=[True, False])
+def getitem(request):
+    yield _wrap_getter(da_getitem, request.param)
+
+
+@pytest.fixture(params=[True, False])
+def getter_nofancy(request):
+    yield _wrap_getter(da_getter_nofancy, request.param)
+
+
+def _check_task(a, b) -> bool:
+    """
+    Check that two tasks (possibly containing nested tasks) are equal, where
+    equality is lax by allowing the callable in a SubgraphCallable to be the same
+    as a non-wrapped task.
+    """
+    if len(a) < 1 or len(a) != len(b):
+        return False
+
+    a_get = (
+        list(a[0].dsk.values())[0][0] if isinstance(a[0], SubgraphCallable) else a[0]
+    )
+    b_get = (
+        list(b[0].dsk.values())[0][0] if isinstance(b[0], SubgraphCallable) else b[0]
+    )
```

Comment:
Yes, I suppose they could be any callables, though I'm only envisioning getters (and made it private since I don't want it to be confused for a more general function).

Happy to make the names clearer.